internal/provider/cluster/cluster.go

```diff
@@ -411,7 +411,7 @@ func resourceClusterDelete(ctx context.Context, d *schema.ResourceData, meta int
 	}
 
 	if !strings.EqualFold(r.Status, "QUEUED") && !strings.EqualFold(r.Status, "IN_PROGRESS") && !strings.EqualFold(r.Status, "COMPLETED") {
-		return diag.Errorf("delete request failure: %q", r.UserFriendlyError)
+		return diag.Errorf("delete request failure, cluster request id: %q", r.ID)
```

Comment:
```suggestion
		return diag.Errorf("delete request failure cluster request id #%d: %q", r.ID, r.ProgressDescription)
```

and please do the same for other cases